src/background.c

```diff
@@ -144,6 +145,60 @@ set_permission (const char *app_id,
     }
 }
 
+/* This is conservative, but lets us avoid escaping most
+   regular Exec= lines, which is nice as that can sometimes
+   cause problems for apps launching desktop files. */
+static gboolean
+need_quotes (const char *str)
+{
+  const char *p;
+
+  for (p = str; *p; p++)
+    {
+      if (!g_ascii_isalnum (*p) &&
+          strchr ("-_%.=:/@", *p) == NULL)
+        return TRUE;
+    }
+
+  return FALSE;
+}
+
+static char *
+maybe_quote (const char *str)
+{
+  if (need_quotes (str))
+    return g_shell_quote (str);
+  return g_strdup (str);
+}
+
+static char **
+rewrite_commandline (const char *app_id,
+                     const char * const *commandline)
```

Comment:
Why is this quoting the commandline, that doesn't seem right. If we pass the commandline to the backend as an argv there is no need to escape anything. Any escaping should be done by the backend.